endpoints/openrtb2/auction.go

```diff
@@ -214,8 +205,27 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 
 	secGPC := r.Header.Get("Sec-GPC")
 
+	//ExtractFPDForBidders removes FPD related structures from request.
+	//In order to preserve original request and return it properly in response.ext.debug.resolvedrequest
+	//original request should be copied before FPD execution
+	resolvedBidReq, err := exchange.DeepCopy(req.BidRequest)
```

Comment:
Good points. I'll fix point 1, didn't think about this optimization. Right, in majority of cases this functionality will not be needed.
I was thinking about point 2 as well. I'll see how much changes it will need